### ISSUE TITLE
`CMAKE_INSTALL_PREFIX` breaks build_ext

### ISSUE THREAD

**Issue by vyasr** [scikit-build/scikit-build#721]
Specifying the `CMAKE_INSTALL_PREFIX` with `build_ext` causes problems for scikit-build. The problem is due to the following properties of scikit-build:
- We parse and forward along all CMake arguments to CMake
- The `python setup.py install` command does not map to CMake installs, it instead uses standard setuptools and some Python functions for the copying required in the installation step.
- CMake `install` commands _are_ invoked as part of the build step to copy from `cmake-build` -> `cmake-install`
- Specifying `CMAKE_INSTALL_PREFIX` ends up affecting the `cmake-build` -> `cmake-install` copy

That last bit is what causes problems. If the install prefix is outside the source directory (which it usually is), we hit [this condition](https://github.com/scikit-build/scikit-build/blob/master/skbuild/setuptools_wrap.py#L796) and fail.

Without deeper knowledge of scikit-build, there are two options that I can think of:
- Either ignore or error out when `CMAKE_INSTALL_PREFIX` is specified. Do we actually need to support this variable for any purpose? Python installation always proceeds using `setup.py install` or `pip install`, and in both cases users only need to install to site-packages or the user local site-packages (using --user).
- Parse out `CMAKE_INSTALL_PREFIX` from the provided CMake arguments and then use it to determine the behavior of `python setup.py install`. This option would make sense if a goal of scikit-build is to support installation to some arbitrary directory similar to what CMake supports.

**Comment by henryiii** [comment]
We should at least filter `CMAKE_INSTALL_PREFIX` from `CMAKE_ARGS`. `SKBUILD_CONFIGURE_ARGS` is quite explicitly only scikit-build, so it should never be set there, but `CMAKE_ARGS` is a condo-forge convention, so more likely to be set. I thought we had ordered it so that our `CMAKE_INSTALL_PREFIX` would override any passed in one, but maybe not?

**Comment by henryiii** [comment]
https://github.com/scikit-build/scikit-build/blob/cc69610e8a0a8d791b70e3bda5fa549c6469f748/skbuild/setuptools_wrap.py#L521-L522

Does that not work?

**Comment by vyasr** [comment]
This works. I'm running into the issue because our wrapper around setup.py allows users to provide CMake arguments and eventually ends up providing the `CMAKE_ARGS` in the environment as part of the setup.py command invocation. Prior to starting this investigation I wasn't aware realize that scikit-build was automatically prepending environment `CMAKE_ARGS` to the user-provided set. Now that I am, my plan is to simply put those user-provided arguments into an `EXTRA_CMAKE_ARGS` variable to avoid shadowing and conflation of the env var with the one created within the script.

Now that I understand the problems a little bit better, the only ask I have to resolve this issue is that we also add some error checking (probably in [`parse_skbuild_args`](https://github.com/scikit-build/scikit-build/blob/master/skbuild/setuptools_wrap.py#L94)) so that if a user specifies `CMAKE_INSTALL_PREFIX` explicitly on the command line with `build_ext` they either get a more obvious error or a warning that it's being ignored. The error that currently gets printed from [here](https://github.com/scikit-build/scikit-build/blob/c003a5672566111643aad33b7fe032394b9541cc/skbuild/setuptools_wrap.py#L790) seems too low-level to me. It took me a bit of debugging to connect that to the command line parameter that was being provided (although in my case we evidently also have a few extra layers of problems...).  Is there _ever_ a reason for a user to specify a custom install directory during the `build_ext` step? That process is entirely managed by scikit-build, right, so there's no reason to specify an alternate prefix? 

I'm happy to contribute that fix if you agree that it makes sense.

**Comment by henryiii** [comment]
`CMAKE_ARGS` already gets provided by conda-forge, so overwriting that would be a bad idea anyway - it's better to use something specific for your package (in case it's added to conda-forge, for example).

`CMAKE_INSTALL_PREFIX` is included in condo-forge's `CMAKE_ARGS`, so it should be fairly silently ignored if it's there (which it is, from what I understand).

I don't think it makes sense to pass it after the `--`, via `cmake_args=`, or via `SKBUILD_CONFIGURE_ARGS`, though since those are explicitly targeting scikit-build, an error is better than being silent about it. If you want to improve that error, that's probably fine.

**Comment by vyasr** [comment]
Right, my guess is that whoever wrote our build scripts initially wasn't aware of the fact that we were shadowing a variable with special meaning for conda-forge. Now that I know that, I can fix that on our end.

Feel free to assign me this issue, I can take care of it when I finally set aside a batch of time to investigate the various scikit-build issues that I opened while transitioning RAPIDS to scikit-build.
